### ISSUE TITLE
MX: Malformed or unexpected name server reply

### ISSUE THREAD

**Issue by macskay** [Mailu/Mailu#1055]
Hi guys, 
I'm having problems sending out mails via smtp. It worked a couple of days ago, now every mail I sent bounces back with a

`MX: Malformed or unexpected name server reply`

I've looked on various search engines and all I could find was something about my MX entry in my DNS Zone file, however the DNS file hasn't changed an is still valid as this entry shows:

```
max@guybrush ~ $ dig @ns1.first-ns.de mx gothax.net
; <<>> DiG 9.11.3-1ubuntu1.8-Ubuntu <<>> @ns1.first-ns.de mx gothax.net
; (2 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 61814
;; flags: qr aa rd; QUERY: 1, ANSWER: 1, AUTHORITY: 3, ADDITIONAL: 2
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;gothax.net.			IN	MX

;; ANSWER SECTION:
gothax.net.		86400	IN	MX	10 mail.gothax.net.

;; AUTHORITY SECTION:
gothax.net.		86400	IN	NS	robotns3.second-ns.com.
gothax.net.		86400	IN	NS	robotns2.second-ns.de.
gothax.net.		86400	IN	NS	ns1.first-ns.de.

;; ADDITIONAL SECTION:
mail.gothax.net.	86400	IN	A	176.9.103.111

;; Query time: 35 msec
;; SERVER: 213.239.242.238#53(213.239.242.238)
;; WHEN: Tue Jun 25 23:51:23 CEST 2019
;; MSG SIZE  rcvd: 174

```

Tests via mxtoolbox.com also show that the Mail Server seems to be setup correctly. I can add new email adresses in the admin backend and also receive emails normally. 

Things I tried so far:

I've tried different versions of Mailu (1.5, 1.6, master) with the same outcome. I tried sending mails via telnet which also gives the same outcome.

I also came across issue #26 where a workaround within the `/etc/resolv.conf` is suggested but that didn't work either.

I recreated the docker containers with docker-compose and tried setting dns server manually for `mailu_smtp_1`. 

I telnetted into the SMTP port and tried sending a message from there, with the same outcome

I put my email adress in thunderbird (which automatically detected the necessary ports) and tried to send a mail from there, with the same outcome.

I disabled ufw for some time as well as flushed my ip tables.

I can't seem to figure out what the problem is here. Has anyone experienced something similar and know an answer to this?

**Comment by kaiyou** [comment]
Does this happen when sending or receiving email? As I understand it it is an issue about sending email and you should check the MX of the recipient address.

**Comment by macskay** [comment]
The problem appears when sending mails yes. The MX records of the recipients are good, as I tried several big companies like gmail, yahoo and some private mail servers which are set up correctly.

**Comment by macskay** [comment]
I've also found out that mailu_resolver_1 which is used as the dns server by all other containers can't actually resolve hosts itself, so pinging any site from within the resolver container does nothing.

**Comment by kaiyou** [comment]
Okay. Could you provide some logs from the SMTP and resolver containers?

``docker-compose logs --tail=1000 resolver smtp ``

**Comment by macskay** [comment]
Alright; I've resent another email to showcase the error again

**smtp**
```
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/qmgr[1557]: 8A33410009F4: from=<mkling@gothax.net>, size=1073, nrcpt=1 (queue active)
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[9531]: 8A33410009F4: to=<klingmannm@gmail.com>, relay=none, delay=23704, delays=23679/0.03/25/0, dsn=4.4.3, status=deferred (Name service error for name=gmail.com type=MX: Malformed or unexpected name server reply)
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[8689]: connect from localhost[127.0.0.1]
postfix/smtpd[8689]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/qmgr[1557]: 74F8C10009EA: from=<mkling@gothax.net>, size=1106, nrcpt=1 (queue active)
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[9929]: 74F8C10009EA: to=<klingmannm@gmail.com>, relay=none, delay=25619, delays=25594/0.04/25/0, dsn=4.4.3, status=deferred (Name service error for name=gmail.com type=MX: Malformed or unexpected name server reply)
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/qmgr[1557]: 5434310009E0: from=<mkling@gothax.net>, size=1108, nrcpt=1 (queue active)
postfix/smtp[10434]: 5434310009E0: to=<klingmannm@gmail.com>, relay=none, delay=29769, delays=29744/0.04/25/0, dsn=4.4.3, status=deferred (Name service error for name=gmail.com type=MX: Malformed or unexpected name server reply)
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[9695]: connect from localhost[127.0.0.1]
postfix/smtpd[9695]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/qmgr[1557]: 8A33410009F4: from=<mkling@gothax.net>, size=1073, nrcpt=1 (queue active)
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[10931]: 8A33410009F4: to=<klingmannm@gmail.com>, relay=none, delay=27904, delays=27879/0.04/25/0, dsn=4.4.3, status=deferred (Name service error for name=gmail.com type=MX: Malformed or unexpected name server reply)
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/qmgr[1557]: 74F8C10009EA: from=<mkling@gothax.net>, size=1106, nrcpt=1 (queue active)
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[11338]: 74F8C10009EA: to=<klingmannm@gmail.com>, relay=none, delay=29819, delays=29794/0.04/25/0, dsn=4.4.3, status=deferred (Name service error for name=gmail.com type=MX: Malformed or unexpected name server reply)
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[11552]: connect from mailu_front_1.mailu_default[192.168.203.7]
postfix/smtpd[11552]: D98F010009F8: client=mailu_webmail_1.mailu_default[192.168.203.11], sasl_method=XCLIENT, sasl_username=mkling@gothax.net
postfix/cleanup[11555]: D98F010009F8: replace: header Received: from [127.0.0.1] (mailu_webmail_1.mailu_default [192.168.203.11])??by mail.gothax.net (Postfix) with ESMTPA id D98F010009F8??for <klingmannm@gmail.com>; Fri, 28 Jun 2019 09:23:08 +0000 (UTC) from mailu_webmail_1.mailu_default[192.168.203.11]; from=<mkling@gothax.net> to=<klingmannm@gmail.com> proto=ESMTP helo=<[127.0.0.1]>: Received: from authenticated-user (PRIMARY_HOSTNAME [PUBLIC_IP])??by mail.gothax.net (Postfix) with ESMTPA id D98F010009F8??for <klingmannm@gmail.com>; Fri, 28 Jun 2019 09:23:08 +0000 (UTC)
postfix/cleanup[11555]: D98F010009F8: message-id=<8a5f7d077693dc83900e21ddc45d2d09@gothax.net>
postfix/qmgr[1557]: D98F010009F8: from=<mkling@gothax.net>, size=1112, nrcpt=1 (queue active)
postfix/smtpd[11552]: disconnect from mailu_webmail_1.mailu_default[192.168.203.11] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[11556]: D98F010009F8: to=<klingmannm@gmail.com>, relay=none, delay=30, delays=5.1/0.02/25/0, dsn=4.4.3, status=deferred (Name service error for name=gmail.com type=MX: Malformed or unexpected name server reply)
postfix/smtpd[10707]: connect from localhost[127.0.0.1]
postfix/smtpd[10707]: disconnect from localhost[127.0.0.1] quit=1 commands=1
```


**resolver**
```
root@gothax /srv/gothax.net/live/mail/app # docker logs mailu_resolver_1
[1561680102] unbound.conf[1:0] error: Could not open logfile /dev/stdout: Permission denied
[1561680102] unbound.conf[1:0] notice: init module 0: validator
[1561680102] unbound.conf[1:0] notice: init module 1: iterator
[1561680102] unbound.conf[1:0] info: start of service (unbound 1.7.3).
```

**Comment by kaiyou** [comment]
Could you try temporarily commenting out the dns settings from the SMTP service in the compose file?

So we at least be certain it is related to dns.

**Comment by macskay** [comment]
Hey @kaiyou,
I commented out the section in the smtp service as such:
```
  smtp:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}postfix:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    volumes:
      - "/srv/gothax.net/live/mail/var/mailu-home/overrides:/overrides"
    depends_on:
      - front
      - resolver
    # dns:
    #   - 192.168.203.254
```
and reconstructed the container. I then sent another email and this is the log:

```
root@gothax /srv/gothax.net/live/mail/app # docker logs -f mailu_smtp_1
rsyslogd 8.34.0: running as pid 1, enabling container-specific defaults, press ctl-c to terminate rsyslog
rsyslogd: environment variable TZ is not set, auto correcting this to TZ=UTC  [v8.34.0 try http://www.rsyslog.com/e/2442 ]
rsyslogd:  [origin software="rsyslogd" swVersion="8.34.0" x-pid="1" x-info="http://www.rsyslog.com"] start
postfix/master[109]: daemon started -- version 3.3.0, configuration /etc/postfix
postfix/smtpd[124]: connect from localhost[127.0.0.1]
postfix/smtpd[124]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[126]: connect from mailu_front_1.mailu_default[192.168.203.3]
postfix/smtpd[126]: 4CB8E1000B07: client=mailu_webmail_1.mailu_default[192.168.203.11], sasl_method=XCLIENT, sasl_username=mkling@gothax.net
postfix/cleanup[141]: 4CB8E1000B07: replace: header Received: from [127.0.0.1] (mailu_webmail_1.mailu_default [192.168.203.11])??by mail.gothax.net (Postfix) with ESMTPA id 4CB8E1000B07??for <klingmannm@gmail.com>; Fri, 28 Jun 2019 10:43:51 +0000 (UTC) from mailu_webmail_1.mailu_default[192.168.203.11]; from=<mkling@gothax.net> to=<klingmannm@gmail.com> proto=ESMTP helo=<[127.0.0.1]>: Received: from authenticated-user (PRIMARY_HOSTNAME [PUBLIC_IP])??by mail.gothax.net (Postfix) with ESMTPA id 4CB8E1000B07??for <klingmannm@gmail.com>; Fri, 28 Jun 2019 10:43:51 +0000 (UTC)
postfix/cleanup[141]: 4CB8E1000B07: message-id=<6855fbc0e23d39830290c17151d559c5@gothax.net>
postfix/qmgr[113]: 4CB8E1000B07: from=<mkling@gothax.net>, size=1084, nrcpt=1 (queue active)
postfix/smtpd[126]: disconnect from mailu_webmail_1.mailu_default[192.168.203.11] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5
postfix/smtpd[124]: connect from localhost[127.0.0.1]
postfix/smtpd[124]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[124]: connect from localhost[127.0.0.1]
postfix/smtpd[124]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtpd[124]: connect from localhost[127.0.0.1]
postfix/smtpd[124]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[146]: connect to gmail-smtp-in.l.google.com[108.177.15.27]:25: Operation timed out
postfix/smtp[146]: connect to gmail-smtp-in.l.google.com[2a00:1450:400c:c0b::1a]:25: Address not available
postfix/smtp[146]: connect to alt1.gmail-smtp-in.l.google.com[2a00:1450:4010:c0f::1a]:25: Address not available
postfix/smtpd[124]: connect from localhost[127.0.0.1]
postfix/smtpd[124]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[146]: connect to alt1.gmail-smtp-in.l.google.com[108.177.14.26]:25: Operation timed out
postfix/smtpd[124]: connect from localhost[127.0.0.1]
postfix/smtpd[124]: disconnect from localhost[127.0.0.1] quit=1 commands=1
postfix/smtp[146]: connect to alt2.gmail-smtp-in.l.google.com[74.125.200.27]:25: Operation timed out
postfix/smtp[146]: 4CB8E1000B07: to=<klingmannm@gmail.com>, relay=none, delay=139, delays=5.2/0.07/134/0, dsn=4.4.1, status=deferred (connect to alt2.gmail-smtp-in.l.google.com[74.125.200.27]:25: Operation timed out)
```

So the `Operation timed out` seems indeed to point to a DNS issue, since the timeout probably happens because of the smtp container not being able to resolve gmail.com.

**Comment by kaiyou** [comment]
Okay, so problem 1 is definitely related to the way your Docker handles external dns. Could you check your Docker version maybe?

Timeouts are probably related to outgoing TCP/25 being blocked by a firewall. Could it be your local firewall? Where is the machine hosted otherwise?

**Comment by macskay** [comment]
The versions are as follows:
```
root@gothax ~ # docker --version
Docker version 18.06.0-ce, build 0ffa825
root@gothax ~ # docker-compose --version
docker-compose version 1.24.0, build 0aa59064
```

My firewall is active but 25/tcp is allowed:
```
root@gothax ~ # ufw status | grep 25
25                         ALLOW       Anywhere                  
25 (v6)                    ALLOW       Anywhere (v6) 
```

Tested with telnet from my local system:
```
max@guybrush ~ $ telnet mail.gothax.net 25
Trying 176.9.103.111...
Connected to mail.gothax.net.
Escape character is '^]'.
220 mail.gothax.net ESMTP ready
```

**Comment by macskay** [comment]
Alriiiight Update Time:

After another hours and hours of headache I found out that NONE of my docker containers could resolve any domains/urls, which led me the an error in my host config, which I fixed after a lot of looking and searching, installing dnsmasq, network-manager and so on. Long story short: It's working now, so it indeed was a problem in my HOSTs DNS rather than in my containers. The resolv.conf is now setting up correctly and all containers can resolve the names.

Thanks anyway! Mailu is back in prod for me!Can thus be closed

**Comment by svaldivia2** [comment]
> Thanks anyway! Mailu is back in prod for me!Can thus be closed

I'm in the same trouble... could you explain what was wrong and how did you solve it?

Thanks in advance

**Comment by undestroyer** [comment]
Faced the same problem. I have no idea what caused it. Server had problems with free disk space and I accidently started anoter instance of nginx at 80 port. Problem was solved by fixing free disk space problem, stoppping new nginx instance, removing docker containers and recreating them with `docker-compose up -d`. May be this information could help debug this issue.

**Comment by PiotrFerenc** [comment]
Remove all images and start using docker-compose up 

```
docker stop $(docker ps -a -q)
docker rm $(docker ps -a -q)
docker-compose up -d
```

**Comment by xteenux** [comment]
faced same issue today, but in my case is type=AAAA: Malformed or unexpected name server reply, 

can anyone help me to check ?

this I attach my smtp and resolver logs

`Attaching to mailu_smtp_1, mailu_resolver_1
resolver_1   | [1587598085] unbound.conf[1:0] error: Could not open logfile /dev/stdout: Permission denied
resolver_1   | [1587598085] unbound.conf[1:0] notice: init module 0: validator
resolver_1   | [1587598085] unbound.conf[1:0] notice: init module 1: iterator
resolver_1   | [1587598085] unbound.conf[1:0] info: start of service (unbound 1.9.1).
resolver_1   | [1587598085] unbound.conf[1:0] info: generate keytag query _ta-4f66. NULL IN
smtp_1       | Apr 22 23:28:25 mail postfix[107]: Postfix is running with backwards-compatible default settings
smtp_1       | Apr 22 23:28:25 mail postfix[107]: See http://www.postfix.org/COMPATIBILITY_README.html for details
smtp_1       | Apr 22 23:28:25 mail postfix[107]: To disable backwards compatibility use "postconf compatibility_level=2" and "postfix reload"
smtp_1       | Apr 22 23:28:25 mail postfix/postfix-script[173]: starting the Postfix mail system
smtp_1       | Apr 22 23:28:25 mail postfix/master[174]: daemon started -- version 3.4.7, configuration /etc/postfix
smtp_1       | Apr 22 23:28:50 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
smtp_1       | Apr 22 23:28:50 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: connect from mailu_front_1.mailu_default[192.168.203.2]
smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: 9F0DDE0A80: client=mailu_webmail_1.mailu_default[192.168.203.11], sasl_method=XCLIENT, sasl_username=bashexp@fhanafi.my.id
smtp_1       | Apr 22 23:28:57 mail postfix/cleanup[190]: 9F0DDE0A80: replace: header Received: from mail.fhanafi.my.id (mailu_webmail_1.mailu_default [192.168.203.11])??by mail.fhanafi.my.id (Postfix) with ESMTPA id 9F0DDE0A80??for <teenux20@gmail.com>; Wed, 22 Apr 2020 23:28:57 +0000 from mailu_webmail_1.mailu_default[192.168.203.11]; from=<bashexp@fhanafi.my.id> to=<teenux20@gmail.com> proto=ESMTP helo=<mail.fhanafi.my.id>: Received: from authenticated-user (PRIMARY_HOSTNAME [PUBLIC_IP])??by mail.fhanafi.my.id (Postfix) with ESMTPA id 9F0DDE0A80??for <teenux20@gmail.com>; Wed, 22 Apr 2020 23:28:57 +0000 (UTC)
smtp_1       | Apr 22 23:28:57 mail postfix/cleanup[190]: 9F0DDE0A80: message-id=<539148980900cc844afba79e37dc641f@fhanafi.my.id>
smtp_1       | Apr 22 23:28:57 mail postfix/qmgr[176]: 9F0DDE0A80: from=<bashexp@fhanafi.my.id>, size=559, nrcpt=1 (queue active)
smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: disconnect from mailu_webmail_1.mailu_default[192.168.203.11] ehlo=2 xclient=0/1 mail=1 rcpt=1 data=1 quit=1 commands=6/7
smtp_1       | Apr 22 23:28:58 mail postfix/smtp[191]: warning: relayhost configuration problem
smtp_1       | Apr 22 23:28:58 mail postfix/smtp[191]: 9F0DDE0A80: to=<teenux20@gmail.com>, relay=none, delay=0.83, delays=0.43/0.21/0.19/0, dsn=4.3.5, status=deferred (Name service error for name=mail.smtp2.go type=AAAA: Malformed or unexpected name server reply)
smtp_1       | Apr 22 23:29:20 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
smtp_1       | Apr 22 23:29:20 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
smtp_1       | Apr 22 23:29:50 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
smtp_1       | Apr 22 23:29:50 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1`

Thanks in advance

**Comment by xteenux** [comment]
faced same issue today, but in my case is type=AAAA: Malformed or unexpected name server reply, 

can anyone help me to check ?

this I attach my smtp and resolver logs

```
Attaching to mailu_smtp_1, mailu_resolver_1
resolver_1   | [1587598085] unbound.conf[1:0] error: Could not open logfile /dev/stdout: Permission denied
resolver_1   | [1587598085] unbound.conf[1:0] notice: init module 0: validator
resolver_1   | [1587598085] unbound.conf[1:0] notice: init module 1: iterator
resolver_1   | [1587598085] unbound.conf[1:0] info: start of service (unbound 1.9.1).
resolver_1   | [1587598085] unbound.conf[1:0] info: generate keytag query _ta-4f66. NULL IN
smtp_1       | Apr 22 23:28:25 mail postfix[107]: Postfix is running with backwards-compatible default settings
smtp_1       | Apr 22 23:28:25 mail postfix[107]: See http://www.postfix.org/COMPATIBILITY_README.html for details
smtp_1       | Apr 22 23:28:25 mail postfix[107]: To disable backwards compatibility use "postconf compatibility_level=2" and "postfix reload"
smtp_1       | Apr 22 23:28:25 mail postfix/postfix-script[173]: starting the Postfix mail system
smtp_1       | Apr 22 23:28:25 mail postfix/master[174]: daemon started -- version 3.4.7, configuration /etc/postfix
smtp_1       | Apr 22 23:28:50 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
smtp_1       | Apr 22 23:28:50 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: connect from mailu_front_1.mailu_default[192.168.203.2]
smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: 9F0DDE0A80: client=mailu_webmail_1.mailu_default[192.168.203.11], sasl_method=XCLIENT, sasl_username=bashexp@fhanafi.my.id
smtp_1       | Apr 22 23:28:57 mail postfix/cleanup[190]: 9F0DDE0A80: replace: header Received: from mail.fhanafi.my.id (mailu_webmail_1.mailu_default [192.168.203.11])??by mail.fhanafi.my.id (Postfix) with ESMTPA id 9F0DDE0A80??for <teenux20@gmail.com>; Wed, 22 Apr 2020 23:28:57 +0000 from mailu_webmail_1.mailu_default[192.168.203.11]; from=<bashexp@fhanafi.my.id> to=<teenux20@gmail.com> proto=ESMTP helo=<mail.fhanafi.my.id>: Received: from authenticated-user (PRIMARY_HOSTNAME [PUBLIC_IP])??by mail.fhanafi.my.id (Postfix) with ESMTPA id 9F0DDE0A80??for <teenux20@gmail.com>; Wed, 22 Apr 2020 23:28:57 +0000 (UTC)
smtp_1       | Apr 22 23:28:57 mail postfix/cleanup[190]: 9F0DDE0A80: message-id=<539148980900cc844afba79e37dc641f@fhanafi.my.id>
smtp_1       | Apr 22 23:28:57 mail postfix/qmgr[176]: 9F0DDE0A80: from=<bashexp@fhanafi.my.id>, size=559, nrcpt=1 (queue active)
smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: disconnect from mailu_webmail_1.mailu_default[192.168.203.11] ehlo=2 xclient=0/1 mail=1 rcpt=1 data=1 quit=1 commands=6/7
smtp_1       | Apr 22 23:28:58 mail postfix/smtp[191]: warning: relayhost configuration problem
smtp_1       | Apr 22 23:28:58 mail postfix/smtp[191]: 9F0DDE0A80: to=<teenux20@gmail.com>, relay=none, delay=0.83, delays=0.43/0.21/0.19/0, dsn=4.3.5, status=deferred (Name service error for name=mail.smtp2.go type=AAAA: Malformed or unexpected name server reply)
smtp_1       | Apr 22 23:29:20 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
smtp_1       | Apr 22 23:29:20 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
smtp_1       | Apr 22 23:29:50 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
smtp_1       | Apr 22 23:29:50 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
```


Thanks in advance

**Comment by xteenux** [comment]
> faced same issue today, but in my case is type=AAAA: Malformed or unexpected name server reply,
> 
> can anyone help me to check ?
> 
> this I attach my smtp and resolver logs
> 
> ```
> Attaching to mailu_smtp_1, mailu_resolver_1
> resolver_1   | [1587598085] unbound.conf[1:0] error: Could not open logfile /dev/stdout: Permission denied
> resolver_1   | [1587598085] unbound.conf[1:0] notice: init module 0: validator
> resolver_1   | [1587598085] unbound.conf[1:0] notice: init module 1: iterator
> resolver_1   | [1587598085] unbound.conf[1:0] info: start of service (unbound 1.9.1).
> resolver_1   | [1587598085] unbound.conf[1:0] info: generate keytag query _ta-4f66. NULL IN
> smtp_1       | Apr 22 23:28:25 mail postfix[107]: Postfix is running with backwards-compatible default settings
> smtp_1       | Apr 22 23:28:25 mail postfix[107]: See http://www.postfix.org/COMPATIBILITY_README.html for details
> smtp_1       | Apr 22 23:28:25 mail postfix[107]: To disable backwards compatibility use "postconf compatibility_level=2" and "postfix reload"
> smtp_1       | Apr 22 23:28:25 mail postfix/postfix-script[173]: starting the Postfix mail system
> smtp_1       | Apr 22 23:28:25 mail postfix/master[174]: daemon started -- version 3.4.7, configuration /etc/postfix
> smtp_1       | Apr 22 23:28:50 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
> smtp_1       | Apr 22 23:28:50 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
> smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: connect from mailu_front_1.mailu_default[192.168.203.2]
> smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: 9F0DDE0A80: client=mailu_webmail_1.mailu_default[192.168.203.11], sasl_method=XCLIENT, sasl_username=bashexp@fhanafi.my.id
> smtp_1       | Apr 22 23:28:57 mail postfix/cleanup[190]: 9F0DDE0A80: replace: header Received: from mail.fhanafi.my.id (mailu_webmail_1.mailu_default [192.168.203.11])??by mail.fhanafi.my.id (Postfix) with ESMTPA id 9F0DDE0A80??for <teenux20@gmail.com>; Wed, 22 Apr 2020 23:28:57 +0000 from mailu_webmail_1.mailu_default[192.168.203.11]; from=<bashexp@fhanafi.my.id> to=<teenux20@gmail.com> proto=ESMTP helo=<mail.fhanafi.my.id>: Received: from authenticated-user (PRIMARY_HOSTNAME [PUBLIC_IP])??by mail.fhanafi.my.id (Postfix) with ESMTPA id 9F0DDE0A80??for <teenux20@gmail.com>; Wed, 22 Apr 2020 23:28:57 +0000 (UTC)
> smtp_1       | Apr 22 23:28:57 mail postfix/cleanup[190]: 9F0DDE0A80: message-id=<539148980900cc844afba79e37dc641f@fhanafi.my.id>
> smtp_1       | Apr 22 23:28:57 mail postfix/qmgr[176]: 9F0DDE0A80: from=<bashexp@fhanafi.my.id>, size=559, nrcpt=1 (queue active)
> smtp_1       | Apr 22 23:28:57 mail postfix/smtpd[188]: disconnect from mailu_webmail_1.mailu_default[192.168.203.11] ehlo=2 xclient=0/1 mail=1 rcpt=1 data=1 quit=1 commands=6/7
> smtp_1       | Apr 22 23:28:58 mail postfix/smtp[191]: warning: relayhost configuration problem
> smtp_1       | Apr 22 23:28:58 mail postfix/smtp[191]: 9F0DDE0A80: to=<teenux20@gmail.com>, relay=none, delay=0.83, delays=0.43/0.21/0.19/0, dsn=4.3.5, status=deferred (Name service error for name=mail.smtp2.go type=AAAA: Malformed or unexpected name server reply)
> smtp_1       | Apr 22 23:29:20 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
> smtp_1       | Apr 22 23:29:20 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
> smtp_1       | Apr 22 23:29:50 mail postfix/smtpd[186]: connect from localhost[127.0.0.1]
> smtp_1       | Apr 22 23:29:50 mail postfix/smtpd[186]: disconnect from localhost[127.0.0.1] quit=1 commands=1
> ```
> 
> Thanks in advance

solving this problem with moving smtp relay to sendgrid, i don't know whats going on with my server, smtp relay trying to resolve AAAA record of smtp2go 😵

**Comment by Yazur-admin** [comment]
I have exactly the same problem, but in my case the mailu_resolver_1 container is "unhealthy".

Let me explain what I did:

I duplicated the "host" VM and changed MAC address, IP address, DNS name.
I did change the IP addresses in the /mailu/docker-compose.yml file.
And changed the DNS name in /etc/hostname and /mailu/mailu.env.

Everything works except the DNS resolution and thus the sending of the mails.

Do you have an idea?

**Comment by matthewbloch** [comment]
My mailu 1.7 system is working fine, but bounces mail with this error if you try to send email to a non-existent domain. (I can see it would also do that if your DNS is borked completely, as per discussions above)

It was confusing enough to send me looking at tcpdump when a user reported it. 

postfix queries for an MX record, A record & AAAA record for the non-existent domain, all of which receive an authoritative "No such name (3)" DNS response. But it logs this bogus "malformed" error after the response to the final AAAA  query. My tcpdump shows the DNS queries & responses as normal, and is pretty good at telling you about bad packets.

And my IPv6 routing is working enough to send to GMail.

So I suspect this error will only show up on mailu hosts with IPv6 enabled, but the postfix build/config _seems_ to be fluffing the AAAA lookup.
